### PR TITLE
pwd: add basic implementation of pwd

### PR DESCRIPTION
**Pull request by xrmx**
The only function implemented is getpwnam

### CODE REVIEW

**Comment by pfalcon** [comment]
Thanks, would be a nice addition! But did you actually test your code? When you say "s" for struct.unpack(), you say "there's a string in this place of the structure", and as you don't specify a length, it's length 1. But struct passwd contains a *pointer* to a string there, not a string. See top comment in py/modstruct.c on how to resolve it.

File: pwd/metadata.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+srctype = micropython-lib
+type = module
+version = 0.0.1
```
Comment (pfalcon) [inline pwd/metadata.txt]: Version should be 0.1, 0.0.x is for completely dummy modules.

File: pwd/pwd.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import ffilib
+import uctypes
+import ustruct as struct
```
Comment (pfalcon) [inline pwd/pwd.py]: Maybe use ustruct directly?

File: pwd/pwd.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from ucollections import namedtuple
+
+
+libc = ffilib.libc()
+
+if libc:
```
Comment (pfalcon) [inline pwd/pwd.py]: Other ffilib-using modules don't have this if, maybe it's not needed here either?

Comment (xrmx) [reply pwd/pwd.py]: I've used os as reference, but will remove

File: pwd/pwd.py (hunk 2 above)
Comment (xrmx) [reply pwd/pwd.py]: fine for me

**Comment by xrmx** [comment]
lol, i've overlooked quite badly the actual output :)

File: pwd/metadata.txt (hunk 1 above)
Comment (xrmx) [reply pwd/metadata.txt]: ok

**Comment by xrmx** [comment]
Updated branch, now the output looks fine, thanks for the review!

File: pwd/pwd.py (hunk 3 above)
Comment (pfalcon) [reply pwd/pwd.py]: I see. Well, I believe there were reports that libc can't be imported e.g. on Windows, maybe on MacOS. So, that "if" was added to not make the module fail on import and just offer constants and things from "uos" module. We don't need such handling in most other modules, so thanks for removing it.

**Comment by pfalcon** [comment]
Merged, thanks! Note that setup.py is autogenerated from metadata.txt: https://github.com/micropython/micropython-lib/wiki/ContributorGuidelines , so that part was tweaked.

**Comment by xrmx** [comment]
Thanks for merging! Completely missed the contribution guidelines :)

**Pull request closed without merging**